runtime/compiler/trj9/runtime/IProfiler.cpp

```diff
@@ -2770,13 +2770,22 @@ TR_IPBCDataCallGraph::setData(uintptrj_t v, uint32_t freq)
          maxWeight = _csInfo._weight[i];
       }
 
-   if (!found && _csInfo._residueWeight >= 0x7FFF)
+   if (!found)
       {
+      // Must update the `residue` bucket
+      uint16_t oldResidueWeight = _csInfo._residueWeight;
+      uint16_t newResidueWeight = oldResidueWeight + freq;
+      if (newResidueWeight > 0x7FFF)
+         newResidueWeight = 0x7FFF;
+      _csInfo._residueWeight = newResidueWeight;
+      returnCount = newResidueWeight;
+
       if (_csInfo._residueWeight > maxWeight)
          {
          if (lockEntry())
             {
-            for (int32_t i = 1; i < NUM_CS_SLOTS; i++)
+            // Reset the entries in reverse order
+            for (int32_t i = NUM_CS_SLOTS - 1; i > 0; i--)
```

Comment:
just curious: why was this change needed?

Reply:
@mpirvu explained to me that it is to try to minimize the chance of a data race with the surrounding code that identifies empty entries looking for zeros. The risk of a race is more of a "we'll misunderstand the profile" than a "we'll crash" risk, but we should still try to minimize its likelihood.

Reply:
Here is an explanation for the reverse order: https://github.com/eclipse/openj9/pull/946#pullrequestreview-90889518

Reply:
Should that explanation be added as a comment on this code?  It'll make it obvious to the next person to read the code...

Reply:
I will add in the comment